### ISSUE TITLE
[CI] testListenersInvokedWhenIndexIsRelocated() failure

### ISSUE THREAD

**Issue by martijnvg** [elastic/elasticsearch#66519]
**Build scan**: https://gradle-enterprise.elastic.co/s/gqtsklf5ff6gs

**Repro line**: `./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.plugins.IndexFoldersDeletionListenerIT.testListenersInvokedWhenIndexIsRelocated" -Dtests.seed=8C2D13C96AAEEC50 -Dtests.security.manager=true -Dtests.locale=et-EE -Dtests.timezone=Africa/Banjul -Druntime.java=11`

**Reproduces locally?**: No

**Applicable branches**: master, 7.x

**Failure history**: https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P7D&search.timeZoneId=Europe/Amsterdam&tests.container=org.elasticsearch.plugins.IndexFoldersDeletionListenerIT&tests.sortField=FAILED&tests.test=testListenersInvokedWhenIndexIsRelocated&tests.unstableOnly=true

(failed a number of times in the last week)

**Failure excerpt**:

```
java.lang.AssertionError: Expecting no shards deleted on node node_t5
-- at __randomizedtesting.SeedInfo.seed([8C2D13C96AAEEC50:42DBF45FC5ADDD9A]:0) |  
  |   | at org.junit.Assert.fail(Assert.java:88) |  
  |   | at org.junit.Assert.assertTrue(Assert.java:41) |  
  |   | at org.elasticsearch.plugins.IndexFoldersDeletionListenerIT.testListenersInvokedWhenIndexIsRelocated(IndexFoldersDeletionListenerIT.java:131)
```

I encountered this failure when running ci on my pr branch.

**Comment by tlrx** [comment]
Thanks Martijn for opening this issue. I just merged bf96bcac42c6e3f2effd053a9ea7cd651f86b06d from66329 that should fix this issue. Looking at your PR (66508) it does not include the fix so it's possible that you hit it again until you merge master (or 7.x) back.

**Comment by martijnvg** [comment]
Thanks @tlrx, I will rebase my branch.
